Improve Qt Introduction (#10)
spyder-ide/spyder-api-docs: docs/qt_introduction.md

Actions can be added to UI elements such as menus and toolbars, and will automatically keep the UI synchronized.
For example, in a word processor, if the user presses a button on the Bold toolbar, the Bold menu item will be automatically checked.

### Dialogs

A Dialog is a GUI component used for brief communications with the user.
Dialogs are commonly used for functions that do not fit into the main interface.
A dialog can be included in an application using the QDialog class.
QDialogs may be modal or modeless.
A modal dialog blocks input to other visible windows in the same application.
The user has to end the interaction with the modal dialog and close it to access to any other window in the application.
For example, dialogs for requesting information or selecting preferences for an application may be modal.
A modeless dialog operates independently of other windows of the same application.
For example, find and replace dialogs allow the user to interact with both the modeless dialog and the application's main window.

> Qt provides some *special dialogs* for the most common use-cases as *file Open/Save*, *font selection*, *error messages*, *color choosing*, *printing*, among others.

Reviewer: ```suggestion
Qt provides some standard dialogs for the most common use cases, such as *file Open/Save*, *error messages*, *color choosing*, *printing*, among others.
```

### Signals & Slots

In the GUI programming is important to provide a way to communicate objects through notifications.
In Qt, signals and slots are used to support this communication.

**Signal** is a notification emitted by a widget when a particular event occurs.
Example of these events includes: pressing a button, changing text in an input box, changing text in the window.
Many signals are initiated by an user action, but this is not a rule.
Qt's widgets have different predermined signals, but you can add your own signals to the widgets.

**Slot** is a function that is invoked in response to a specific signal.
Functions or methods could be used as slots, by connecting a signal to them.

> If a signal sends data, the receiver callable will also receive it.

Reviewer: ```suggestion
If a signal sends data, the slot function will be called with it.
```
